src/archivist/filter/index.js

```diff
@@ -142,17 +142,18 @@ export function convertRelativeURLsToAbsolute(document, baseURL) {
 // Works in place
 function remove(webPageDOM, noiseSelectors) {
   const rangeSelections = [];
+  const nodes = [];
 
   [].concat(noiseSelectors).forEach(selector => {
     if (typeof selector === 'object') {
       rangeSelections.push(selectRange(webPageDOM, selector));
     } else {
-      Array.from(webPageDOM.querySelectorAll(selector)).forEach(node => node.classList.add('open-terms-archive-remove-marker')); // Tag the nodes to be deleted instead of deleting them directly. Ensure all nodes are still present on the page when trying to use precedent-dependent selector (`~`, `+`, `:nth-child`…)
+      Array.from(webPageDOM.querySelectorAll(selector)).forEach(node => nodes.push(node));
```

Comment:
Avoid unnecessary iteration 🌱 ⚡ 
```suggestion
      nodes.push(...webPageDOM.querySelectorAll(selector));
```